process/unDelegateValidators.go

```diff
@@ -0,0 +1,124 @@
+package process
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/multiversx/mx-chain-tools-accounts-manager-go/core"
+	"github.com/multiversx/mx-chain-tools-accounts-manager-go/data"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+const (
+	maxNumberOfParallelRequests   = 40
+	getUnStakedTokensListEndpoint = "getUnStakedTokensList"
+)
+
+func (ag *accountsGetter) putUndelegatedValuesFromValidatorsContract(accountsWithStake map[string]*data.AccountInfoWithStakeValues) error {
+	if ag.validatorsContract == "" {
+		return nil
+	}
+
+	unDelegatedValues, err := ag.getUnDelegatedValuesFromValidatorsContract(accountsWithStake)
+	if err != nil {
+		return err
+	}
+
+	for address, unDelegatedValue := range unDelegatedValues {
+		if _, found := accountsWithStake[address]; !found {
+			continue
+		}
+
+		accountsWithStake[address].UnDelegateValidator = unDelegatedValue.String()
+		accountsWithStake[address].UnDelegateValidatorNum = core.ComputeBalanceAsFloat(unDelegatedValue.String())
+	}
+
+	return nil
+}
+
+func (ag *accountsGetter) getUnDelegatedValuesFromValidatorsContract(accountsWithStake map[string]*data.AccountInfoWithStakeValues) (map[string]*big.Int, error) {
+	defer logExecutionTime(time.Now(), "Fetched undelegated values from validators contract")
+
+	unDelegatedValue := make(map[string]*big.Int)
+
+	done, wg := make(chan struct{}, maxNumberOfParallelRequests), &sync.WaitGroup{}
+	errors := make([]string, 0)
+	for address := range accountsWithStake {
+		done <- struct{}{}
+		wg.Add(1)
+
+		go func(addr string) {
+			value, err := ag.getUnDelegatedValueForAddressValidatorsContract(addr, done, wg)
```

Comment:
why not move here the defer from getUnDelegatedValueForAddressValidatorsContract, this way channel and waitGroup won't be needed there + easier to read